pulser-core/pulser/json/abstract_repr/serializer.py

```diff
@@ -49,6 +49,9 @@ def default(self, o: Any) -> dict[str, Any] | list | int:
         elif isinstance(o, set):
             return list(o)
         elif isinstance(o, complex):
+            if not o.imag:
+                # Try to return a real number when possible
+                return o.real
```

Comment:
isn't this the wrong way around?

Reply:
How so @awennersteen ? If there is no imaginary part, we return just the real part

Reply:
I think I added a edit at the same time as you commented

Reply:
If you insist I can be explicit but `o.imag == 0` is the only case that will enter this `if`

Reply:
I don't insist , but thanks

It really confused me as I at first assumed you wanted to always throw away the imaginary part